### PR TITLE
xds: add config for pick_first LB policy extension

### PR DESCRIPTION
**Pull request by easwars**
This PR adds an LB policy extension configuration, to be used in gRPC via the new `load_balancing_policy` field:

* `pick_first`: This allows configuring the built-in PICK_FIRST LB policy via the new extension-based API.

Risk Level: Low
Testing: N/A
Docs Changes: Included in PR
Release Notes: N/A
Platform Specific Features: N/A

CC @markdroth

### CODE REVIEW

File: api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/pick_first/v3;pick_firstv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Pick First Load Balancing Policy]
+// [#extension: envoy.load_balancing_policies.pick_first]
+
+// This configuration allows the built-in PICK_FIRST LB policy to be configured via the LB policy
+// extension point. See the :ref:`load balancing architecture overview
+// <arch_overview_load_balancing_types>` for more information.
```
Comment (mattklein123) [inline api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto:18]: Is this going to get implemented in Envoy? Do you have these docs ready? Are you going to submit the code in the final PR? I don't have any context on this, thank you.

/wait-any

Comment (easwars) [reply api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto:18]: No, I don't expect this to be implemented in Envoy. We do have a `pick_first` LB policy in gRPC and we want the ability to configure this policy as the leaf policy via the custom LB policy mechanism in xDS. I do have a [document](https://github.com/grpc/proposal/pull/357) out for the changes we are planning to make to the `pick_first` LB policy which covers this.

@markdroth: Do you want to add anything?

Comment (mattklein123) [reply api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto:18]: OK then please remove all the docs and just add the not-implemented-hide annotation. Then you can remove most of the other changes. Thank you.

/wait

Comment (markdroth) [reply api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto:18]: Just to confirm, this is a policy we have in gRPC, but this particular policy is unlikely to be implemented in Envoy.

Adding the not-implemented-hide annotation and removing the Envoy documentation sounds good.

Comment (easwars) [reply api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto:18]: Done. Thanks.

Comment (mattklein123) [reply api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto:18]: Can you please remove the arch_voerview stuff also? It's not real. Just have the not-implemented-hide annotation and perhaps a short note that this is currently used in gRPC only. Thanks.

/wait

Comment (easwars) [reply api/envoy/extensions/load_balancing_policies/pick_first/v3/pick_first.proto]: Done. Thanks.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**